7-extract_plier_pathways.R

```diff
@@ -112,28 +150,43 @@ for(seed_index in 1:length(norm.train.files)) {
   doParallel::registerDoParallel(cl)
   
   # at each titration level (0-100% RNA-seq)
-  perc_seq <- as.character(seq(0, 100, 10))
-  norm_methods <- c("log", "npn", "qn", "tdm", "z")
-  plier_results_list <- foreach(ps = perc_seq, .packages = c("PLIER", "doParallel"), .export = c("check_all_same")) %dopar% {
-    foreach(nm = norm_methods, .packages = c("PLIER", "doParallel"), .export = c("check_all_same")) %dopar% {
+  #perc_seq <- as.character(seq(0, 100, 10))
+  #norm_methods <- c("log", "npn", "qn", "tdm", "z")
+  perc_seq <- as.character(seq(0, 100, 50))
+  norm_methods <- c("log", "qn")
+  plier_results_list <- foreach(ps = perc_seq,
+                                .packages = c("PLIER", "doParallel")) %dopar% {
+    foreach(nm = norm_methods) %dopar% {
       
       if (nm %in% names(norm.train.list[[ps]])) {
-        if(any(apply(norm.train.list[[ps]][[nm]], 1, check_all_same))) {
-          c("Some rows all same value...") # TODO This shouldn't be happening?
-        } else {
-          # minimum k for PLIER = 2*num.pc
-          # TODO alternatively, should we just set one k for all data sets?
-          set.k <- 2*PLIER::num.pc(norm.train.list[[ps]][[nm]][common.genes, ])
-          
-          # PLIER main function
-          PLIER::PLIER(as.matrix(norm.train.list[[ps]][[nm]][common.genes, ]),
-                       all.paths[common.genes, ],
-                       k = set.k,
-                       trace = FALSE,
-                       scale = TRUE)
+        
+        # remove any rows with all the same value
+        # TODO this may now be superfluous given fixed 0-1 rescaling issue
+        all.same.indx <- which(apply(norm.train.list[[ps]][[nm]], 1,
+                                     check_all_same))
+        if (length(all.same.indx) > 0) {
+          norm.train.list[[ps]][[nm]] <- norm.train.list[[ps]][[nm]][-all.same.indx, ]
         }
+        
+        # get common genes
+        common.genes <- PLIER::commonRows(all.paths,
+                                          norm.train.list[[ps]][[nm]])      
```

Comment:
This move makes sense to me if one was to drop the all same genes. However, if it was a large number of genes you might imagine that it would impact certain pathways inclusion in the model. I know we've discussed moving away from this evaluation, so it may be neither here nor there.

Reply:
I expect `common.genes` to be highly similar across data sets, with only an occasional gene different, especially after resolution of the zero_to_one transformation issue.